apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -6,28 +6,209 @@
 package v2alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// ResourceName is the name of a Deployment Component
+type ResourceName string
+
+const (
+	// NodeAgentResourceName is the name of the Datadog Agent
+	NodeAgentResourceName ResourceName = "nodeAgent"
+	// ClusterAgentResourceName is the name of the Cluster Agent
+	ClusterAgentResourceName ResourceName = "clusterAgent"
+	// ClusterCheckRunnerResourceName is the name of the Cluster Check Runner
+	ClusterCheckRunnerResourceName ResourceName = "clusterCheckRunner"
```

Comment:
```suggestion
	ClusterChecksRunnerResourceName ResourceName = "clusterChecksRunner"
```

(to be consistent with what we do in Helm https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L1169)